packages/changed-elements-react/src/widgets/V2Widget/componets/VersionCompareSelectModal.tsx

```diff
@@ -0,0 +1,359 @@
+    startChangesetId: args.targetVersion.changesetId as string,
+    endChangesetId: args.currentVersion.changesetId as string,
+  });
+  if (comparisonJob.status === "Completed" && comparisonJob.comparison && comparisonJob.comparison.href) {
+    void runManagerStartComparisonV2({
+      comparisonJob: { comparisonJob: comparisonJob },
+      comparisonJobClient: args.comparisonJobClient,
+      iModelConnection: args.iModelConnection,
+      targetVersion: args.targetVersion,
+      currentVersion: args.currentVersion,
+    });
+    return;
+  }
+  const toastProgressingInterval = setInterval(() => {
+    if (comparisonJob.status !== "Completed" && !args.getDialogOpen()) {
+      toastComparisonJobProcessing(args.currentVersion, args.targetVersion);
+    }
+  }, 8000); // toast progress every 8 seconds
+  if (comparisonJob.status !== "Completed" && comparisonJob.status !== "Error") {
+    toastComparisonJobProcessing(args.currentVersion, args.targetVersion);
+  }
+  while (comparisonJob.status !== "Error") {
+    await new Promise((resolve) => setTimeout(resolve, 5000)); // run loop every 5 seconds
+    if (VersionCompare.manager?.isComparing) {
+      clearInterval(toastProgressingInterval);
+      return;
+    }
+    comparisonJob = (await postOrGetComparisonJob({
+      changedElementsClient: args.comparisonJobClient,
+      iTwinId: args.iModelConnection?.iTwinId as string,
+      iModelId: args.iModelConnection?.iModelId as string,
+      startChangesetId: args.targetVersion.changesetId as string,
+      endChangesetId: args.currentVersion.changesetId as string,
+    })).comparisonJob;
+
+    if (comparisonJob.status === "Completed" && comparisonJob.comparison && comparisonJob.comparison.href) {
+      clearInterval(toastProgressingInterval);
+      if (!args.getDialogOpen() && !VersionCompare.manager?.isComparing) {
+        toastComparisonJobComplete({
+          comparisonJob: { comparisonJob: comparisonJob },
+          comparisonJobClient: args.comparisonJobClient,
+          iModelConnection: args.iModelConnection,
+          targetVersion: args.targetVersion,
+          currentVersion: args.currentVersion,
+        });
+      }
+      return;
+    }
+  }
+};
+
+type PostOrGetComparisonJobParams = {
+  changedElementsClient: IComparisonJobClient;
+  iTwinId: string;
+  iModelId: string;
+  startChangesetId: string;
+  endChangesetId: string;
+};
+
+async function postOrGetComparisonJob(args: PostOrGetComparisonJobParams): Promise<ComparisonJob> {
+  let result: ComparisonJob;
+  try {
+    result = await args.changedElementsClient.postComparisonJob({
+      iTwinId: args.iTwinId,
+      iModelId: args.iModelId,
+      startChangesetId: args.startChangesetId,
+      endChangesetId: args.endChangesetId,
+      headers: { "Content-Type": "application/json" },
+    });
+  } catch (error: unknown) {
+    if (error && typeof error === "object" && "code" in error && error.code !== "ComparisonExists") {
```

Comment:
I don't want to catch the error if it occurs in the get request. The caller of this method should. We only catch the post because a comparison could exist and if it does we then get it.